clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSContext.scala

```diff
@@ -62,6 +73,37 @@ object LakeFSContext {
       .add(StructField("etag", StringType))
       .add(StructField("last_modified", TimestampType))
       .add(StructField("size", LongType))
+      .add(StructField("range_id", StringType))
     spark.createDataFrame(rdd, schema)
   }
+
+  def newDFFromStorageNamespace(
+      spark: SparkSession,
+      storageNamespace: String
+  ): DataFrame = {
+    newDF(spark, "", storageNamespace, "", classOf[LakeFSRepositoryInputFormat])
+  }
+
+  def newDF(
+      spark: SparkSession,
+      repoName: String
+  ): DataFrame = {
+    newDF(spark, repoName, "", "", classOf[LakeFSRepositoryInputFormat])
+  }
+
+  def newDF(
+      spark: SparkSession,
+      repoName: String,
+      commitID: String
+  ): DataFrame = {
+    newDF(spark, repoName, "", commitID, classOf[LakeFSCommitInputFormat])
+  }
+
+  def newRDD(
```

Comment:
Why do we need to create RDDs _and_ DataFrames?

Reply:
It is used in List.scala- not necessarily with good justification (also - you made me realize I had a bug here in the method signature, now fixes).